### PR TITLE
fix(submission): Popping multiple navigation children results results in broken footer state (closes #1463)

### LINKED ISSUE #1463: Popping multiple navigation children (iOS 14+ feature) results in broken footer state
## Avoid duplicates

* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is not already reported in another issue

## Describe the bug

When using the iOS 14+ navigation flow feature of closing multiple child views at once using force touch on the "Back" button of the navigation bar, the footer view moves to a broken state and prevent further inputs until the view is manually closed and reopened. This is only an issue for the submission view controller and can be seen in this video:

https://www.dropbox.com/s/fpqcyulaj1uhoqp/CWA_iOS_Submission_Nav_Bug_Repro.mov?dl=0 

## Expected behaviour

The submission footer view should be activated again, once a child view controller is closed.

## Steps to reproduce the issue

1. Go to "Learn More and Help"
2. Click on the "Next" button
3. Select one of the child view controllers, e.g. "TAN" or "Request TAN"
4. Long-press on the "Back" button in the navigation bar, which brings you back to the info screen
5. Notice that the "Next" button is still grayed out 

## Technical details

- iOS Version: 12.1
- Device: Simulator + iPhone 11 Pro

## Possible Fix

Clean up the footer state by using the UIViewController life cycle delegate `didMoveToParentViewController`.

## Additional context

The issue is fixed in the following PR: https://github.com/corona-warn-app/cwa-app-ios/pull/1464

---
Internal Tracking ID: EXPOSUREAPP-3617

### PR DESCRIPTION
**Pull request by hansemannn**
## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed.
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

## Description

When using the iOS 14+ navigation flow feature of closing multiple child views at once using force touch on the "Back" button of the navigation bar, the footer view moves to a broken state and prevent further inputs until the view is manually closed and reopened. This is only an issue for the submission view controller and can be seen in this video:

https://www.dropbox.com/s/fpqcyulaj1uhoqp/CWA_iOS_Submission_Nav_Bug_Repro.mov?dl=0

### CODE REVIEW

**Review by KaiTeuber — CHANGES_REQUESTED** [review]
Thank you, but this fix only solves the problem at this specific point. We better adjust this in general in EANavigationControllerWithFooter.

**Comment by nickguendling** [comment]
To make this work for navigation between all view controllers I think it would be best to override `popToViewController(...)` in the `ENANavigationControllerWithFooter` extension that already overrides `popViewController(animated:)` like this:

`override func popToViewController(_ viewController: UIViewController, animated: Bool) -> [UIViewController]? {
		let viewController = super.popToViewController(viewController, animated: animated)
		transitionFooterView(to: topViewController)
		return viewController
	}`

What do you think?

**Comment by hansemannn** [comment]
Thanks @KaiTeuber and @nickguendling! Your solution is more generic - great idea! I adjusted the PR to revert the previous change and adopt the other life cycle delegate

**Review by KaiTeuber — APPROVED** [review]
Looking good, thanks

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/NavigationControllerWithFooter/ENANavigationControllerWithFooter.swift [hunk 1]
```diff
@@ -67,6 +67,12 @@ class ENANavigationControllerWithFooter: UINavigationController {
 		updateAdditionalSafeAreaInsets()
 		layoutFooterView()
 	}
+
+	override func popToViewController(_ viewController: UIViewController, animated: Bool) -> [UIViewController]? {
+		let viewController = super.popToViewController(viewController, animated: animated)
+		transitionFooterView(to: topViewController)
+		return viewController
+	}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/NavigationControllerWithFooter/ENANavigationControllerWithFooter.swift:75]: Could you move this into the extension starting in line 143?

**Review by nickguendling — CHANGES_REQUESTED** [review]

**Review by nickguendling — APPROVED** [review]

**Pull request merged**